mesonbuild/dependencies/base.py

```diff
@@ -609,11 +609,20 @@ def _call_pkgbin_real(self, args, env):
         return rc, out
 
     def _call_pkgbin(self, args, env=None):
+        # Always copy the environment since we're going to modify it
+        # with pkg-config variables
         if env is None:
-            fenv = env
-            env = os.environ
+            env = os.environ.copy()
         else:
-            fenv = frozenset(env.items())
+            env = env.copy()
+
+        if self.want_cross:
+            extra_paths = self.env.coredata.get_builtin_option('cross_pkg_config_path')
+        else:
+            extra_paths = self.env.coredata.get_builtin_option('pkg_config_path')
+        env['PKG_CONFIG_PATH'] = ':'.join([os.path.expanduser(os.path.expandvars(p))
```

Comment:
Hmm, do we want to expand env vars and `~` like this? We don't do that for any other var afaik?

Reply:
I guess we can drop it, but it is annoying to have to expand those ourselves.

Reply:
Won't the shell auto-expand it for us unless the user explicitly quotes it with `'`?

Reply:
I'm trying to remember for sure, but I think I added that because my shell wasn't in some cases. I'll double check that

Reply:
Doesn't seem to be necessary, so I've removed them.